megamek/src/megamek/common/EnhancedTabbedPane.java

```diff
@@ -0,0 +1,2354 @@
+/*
+ * Copyright (C) 2025 The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMekLab.
+ *
+ * MegaMekLab is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License (GPL),
+ * version 2 or (at your option) any later version,
+ * as published by the Free Software Foundation.
+ *
+ * MegaMekLab is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * A copy of the GPL should have been included with this project;
+ * if not, see <https://www.gnu.org/licenses/>.
+ *
+ * NOTICE: The MegaMek organization is a non-profit group of volunteers
+ * creating free software for the BattleTech community.
+ *
+ * MechWarrior, BattleMech, `Mech and AeroTech are registered trademarks
+ * of The Topps Company, Inc. All Rights Reserved.
+ *
+ * Catalyst Game Labs and the Catalyst Game Labs logo are trademarks of
+ * InMediaRes Productions, LLC.
+ */
+package megamek.common;
+
+import java.awt.*;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.awt.event.HierarchyEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseMotionAdapter;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowListener;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.swing.*;
+import javax.swing.event.MouseInputAdapter;
+
+import megamek.client.ui.swing.GUIPreferences;
+
+/**
+ * @author Drake
+ *         <p>
+ *         Enhanced tabbed pane with closable, draggable tabs and additional
+ *         action buttons. This component extends
+ *         JTabbedPane to add the following features:
+ *         <ul>
+ *         <li>Closable tabs with an X button on each tab</li>
+ *         <li>Ability to drag and reorder tabs</li>
+ *         <li>Action buttons on the right side of the tab area (sticky and
+ *         not)</li>
+ *         <li>Detachable tabs that can be dragged out (or right-click) into
+ *         floating windows</li>
+ *         </ul>
+ */
```

Comment:
we are not limiting javadoc to 80 columns, you can go wild (120 columns or 160... dont remember)